### PR TITLE
Fix #1289

### LINKED ISSUE #1289: Improvements of iS//iSj
- [x]  Add `vsz` and `idx` in iSj
- [x]  Get Sections Entropy in iS/iSj
- [x]  Get Sections md5/sha1 in iS/iSj
- [x]  Possibility to do rabin2 -K entropy,md5,sha1 -jS $FILE // To get both md5,sha1 and entropy in the same command

-----

- [ ]  Get those informations on the Web-UI

It could be nice to get common hashing of Sections (md5,sha1) with or without an option in `iS` and `iSj` as well as entropy to quickly compare malicious Sections and quickly see if a Section is packed like already existing tools/website (VirusTotal, AVCaesar...)

**VirusTotal**

pe_section_vt

**AVCaesar**

pe_section_avcaesar

### PR DESCRIPTION
**Pull request by alvarofe**
This try fix #1289 though not completely. 

I've added to iS//iSj entropy and md5 without the possibility to choose algorithm. We can define another command (discussion). 

Now rabin2 -K entropy,md5,sha1,sha256 -S[j,q] /bin/ls works

I didn't touch nothing related with the webui. @radare can fix this quickly than me.

The test can be broken and I think I just spotted an issue with entropy I get now `01bebebe` . I'll review tonight

### CODE REVIEW

File: libr/core/bin.c [hunk 1]
```diff
@@ -1294,34 +1294,72 @@ static int bin_sections(RCore *r, int mode, ut64 laddr, int va, ut64 at, const c
...
+						i < sizeof (tmp) - 1; i++)
+						tmp[i] = *ptr++;
+					tmp[i] = '\0';
+					chkstr = r_hash_to_string (NULL, tmp, data, datalen);
+					if (!chkstr || !*chkstr) {
+						if (*ptr && *ptr == ',')
+							ptr++;
+						continue;
+					}
+					aux = malloc (strlen (chkstr)+strlen (tmp)+3);
```
Comment (jvoisin) [inline libr/core/bin.c]: What about `r_str_newf("%s%s", chkstr, aux);` instead?

**Comment by jvoisin** [comment]
I seems that you duped 3 times to (almost) same code :/

**Comment by Maijin** [comment]
Nop, rabin2 -K entropy,md5,sha1,sha256 -S[j,q] /bin/ls doesn't work:

`idx=29 vaddr=0x00400000 paddr=0x00000000 sz=64 vsz=64 perm=m-rw- entropy,md5,sha1,sha256=e3b0c44298fc1c149afbf4c8996fb924 name=ehdr`

Only display md5 but says sha256=

**Comment by alvarofe** [comment]
@Maijin i get this

````
{
         "paddr" : 0,
         "flags" : "m-rw-",
         "name" : "12.__common",
         "size" : 140,
         "vaddr" : 4294989264,
         "sha256" : "c500e44de36ba9e3f5088ede00e79fdae6b5038582a5df40a910f276e9fa8c62",
         "vsize" : 140,
         "md5" : "d107c377700bc1a09172b15b584a3edb",
         "sha1" : "dc58fcb4ed46221844e5be2d450493603e57a5bd",
         "entropy" : "026872a4"
````

**Comment by alvarofe** [comment]
@jvoisin I know I have to move that to a function or something I wanted to know if all of you want iS with paramater to specify the algorithm and see which tests are broken. I wrote this in 20 min with ctrl+c, ctrl+v xD

I'll review the r_str_newf ;)

**Comment by jvoisin** [comment]
Great :)

**Comment by alvarofe** [comment]
enjoy 
https://asciinema.org/a/d7hx9h8c9k58eqglat074ntc6

**Comment by alvarofe** [comment]
if there is some issue tell me ;)

**Comment by alvarofe** [comment]
pr in r2r to fix the issue

**Comment by Maijin** [comment]
Not ready to merge, <empty> should also produce an hash

**Comment by alvarofe** [comment]
updated let's see travis

**Comment by radare** [comment]
Awesome! merged

maybe this can be a problem when a binary or debugger contains a HUGE section (or invalid), so maybe we want a variable to disable that stuff as we already do when loading the bin to avoid too much time spent calculating hashes

**Comment by radare** [comment]
rabin2 not touched, right? https://github.com/radare/radare2/issues/1289

**Comment by alvarofe** [comment]
yeah. rabin2 -K md5,sha1,sha256 -Sj /bin/ls should works

**Pull request closed without merging**